knowledgebase/python-3-9-clickhouse-connect-example.md

```diff
@@ -0,0 +1,114 @@
+---
+Date: 2023-05-08
+---
+
+# A Python 3.9 working example for connecting to ClickHouse Cloud Service
+
+Using the [clickhouse-connect](https://github.com/ClickHouse/clickhouse-connect) driver and python 3.9.
+
+
+check Python version:
+
+```
+$  python -V
+Python 3.9.6
+```
+
+in this example we'll assemble the project in a folder called `ch-python`:
+
+```
+$ mkdir ch-python
+$ cd ch-python
+```
+
+create a dependencies file named `requirements.txt` with:
+
+```
+clickhouse-connect==0.5.23
+urllib3==1.26.15 #this is to workaround dependency hell related to ImportError: urllib3 v2.0 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with LibreSSL 2.8.3. See: https://github.com/urllib3/urllib3/issues/2168
```

Comment:
Where is SSL breaking?  If this is a common problem we should change the clickhouse-connect requirement to exclude urllib 2.0.

Reply:
See: https://github.com/urllib3/urllib3/issues/2168 

without pinning urllib3 to `1.26.15` that the code throws

```
(venv) ➜  three python main.py                 
Unable to connect ClickHouse Connect C to Numpy API [No module named 'numpy'], falling back to pure Python
Traceback (most recent call last):
  File "/opt/cases/00003695_/three/main.py", line 1, in <module>
    import clickhouse_connect
  File "/opt/cases/00003695_/three/venv/lib/python3.9/site-packages/clickhouse_connect/__init__.py", line 1, in <module>
    from clickhouse_connect.driver import create_client
  File "/opt/cases/00003695_/three/venv/lib/python3.9/site-packages/clickhouse_connect/driver/__init__.py", line 9, in <module>
    from clickhouse_connect.driver.httpclient import HttpClient
  File "/opt/cases/00003695_/three/venv/lib/python3.9/site-packages/clickhouse_connect/driver/httpclient.py", line 9, in <module>
    from urllib3 import Timeout
  File "/opt/cases/00003695_/three/venv/lib/python3.9/site-packages/urllib3/__init__.py", line 38, in <module>
    raise ImportError(
ImportError: urllib3 v2.0 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with LibreSSL 2.8.3. See: https://github.com/urllib3/urllib3/issues/2168
```

Reply:
What system is this on?

Reply:
```
➜  ~ uname -a
Darwin clickhousebook.local 22.3.0 Darwin Kernel Version 22.3.0: Mon Jan 30 20:38:37 PST 2023; root:xnu-8792.81.3~2/RELEASE_ARM64_T6000 arm64
```

Reply:
➜  ~ python -V
Python 3.9.6

Reply:
@genzgd I've pushed changes. renamed it only in the last [commit](https://github.com/ClickHouse/clickhouse-docs/pull/1126/commits/806b2f7775d01a227d6078fd41a61f6b30d1cbd0) to take version out of the filename, with minor rewording of the intro